python_tsp/exact/branch_and_bound/solver.py

```diff
@@ -0,0 +1,47 @@
+from typing import List, Tuple
+
+import numpy as np
+
+from python_tsp.exact.branch_and_bound import Node, PriorityQueue
+
+
+def solve_tsp_branch_and_bound(
+    distance_matrix: np.ndarray,
+) -> Tuple[List[int], float]:
+    """
+    Solve the Traveling Salesperson Problem (TSP) using
+    the Branch and Bound algorithm.
+
+    Parameters
+    ----------
+    distance_matrix : numpy.ndarray
+        The distance matrix representing the distances between cities.
+
+    Returns
+    -------
+    Tuple[List[int], float]
+        A tuple containing the optimal path (list of city indices) and its
+        total cost. If the TSP cannot be solved, an empty path and cost of 0
+        will be returned.
+    """
+    num_cities = len(distance_matrix)
+    cost_matrix = np.copy(distance_matrix)
+    inf = np.iinfo(cost_matrix.dtype).max
```

Comment:
From what I can get from [the documentation](https://numpy.org/doc/stable/reference/generated/numpy.iinfo.html), this method only works for integers. 

It is mentioned in the README that the solvers in this library can handle matrices with floats, but I think with this line the function would fail if the distance matrix is not of integer type.

Can your code work floating point numbers as well? If so, maybe replacing this `inf` with the regular `inf` from the `math` module would be enough.

If not, tell me and we can figure something out.

Reply:
Totally right @fillipe-gsm, it only worked for integers, I've fixed this behavior and now it works for distance matrix with float numbers too.